Sources/Swift/Tools/SentryLogBatcher.swift

```diff
@@ -0,0 +1,27 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+@objc
+@_spi(Private) public protocol SentryLogBatcherDelegate {
+    func send(_ envelope: SentryEnvelope)
+}
+
+@objc
+@objcMembers
+@_spi(Private) public class SentryLogBatcher: NSObject {
+    private weak var delegate: (SentryLogBatcherDelegate)?
+    
+    @_spi(Private) public init(delegate: SentryLogBatcherDelegate) {
+        self.delegate = delegate
+        super.init()
+    }
+    
+    @_spi(Private) public func processLog(_ log: SentryLog, with scope: Scope) {
+        do {
+            let envelope = try SentryEnvelope(logs: [log])
```

Comment:
For now we only create the envelope from logs. Actual batching will be implemented in a later PR.